eCommerce platforms/Adobe Magento/src/code/local/Microsoft/Dfp/Model/Observer.php

```diff
@@ -0,0 +1,251 @@
+	{
+		$_metadata = array(
+			"trackingId"		=> $this->dfp->GUID(),
+			"merchantTimeStamp"	=> date('c')
+		);
+
+		// Device FingerPrinting Details
+		$deviceContext = array(
+			"deviceContextId"	=> Mage::getSingleton('core/session')->getFptDfpSessionId(),
+			"ipAddress"			=> getenv('HTTP_CLIENT_IP') ?: getenv('HTTP_X_FORWARDED_FOR') ?: getenv('HTTP_X_FORWARDED') ?: getenv('HTTP_FORWARDED_FOR') ?: getenv('HTTP_FORWARDED') ?: getenv('REMOTE_ADDR'),
+			"provider"			=> "DFPFingerPrinting",
+		);
+
+		//Address details
+		$AddressList = array();
+		$billingAddress = $customer->getDefaultBillingAddress();
+		$billingAddressStreet = $billingAddress->getStreet();
+		$AddressList[] = array(
+			"type"		=> 	"BILLING",
+			"firstName"	=>	$billingAddress["firstname"],
+			"lastName"	=>	$billingAddress["lastname"],
+			"street1"	=>	$billingAddressStreet[0],
+			"street2"	=>	count($billingAddressStreet) > 1 ? $billingAddressStreet[1] : null,
+			"city"		=>	$billingAddress["city"],
+			"state"		=>	$billingAddress["region"],
+			"zipCode"	=>	$billingAddress["postcode"],
+			"country"	=>	$billingAddress["country_id"]
+		);
+
+		$shippingAddress = $customer->getDefaultShippingAddress();
+		$shippingAddressStreet = $shippingAddress->getStreet();
+		$AddressList[] = array(
+			"type"		=> 	"SHIPPING",
+			"firstName"	=>	$shippingAddress["firstname"],
+			"lastName"	=>	$shippingAddress["lastname"],
+			"street1"	=>	$shippingAddressStreet[0],
+			"street2"	=>	count($shippingAddressStreet) > 1 ? $shippingAddressStreet[1] : null,
+			"city"		=>	$shippingAddress["city"],
+			"state"		=>	$shippingAddress["region"],
+			"zipCode"	=>	$shippingAddress["postcode"],
+			"country"	=>	$shippingAddress["country_id"]
+		);
+
+		foreach ($customer->getAdditionalAddresses() as $item) {
+			$AddressList[] = array(
+				"type"		=> 	"SHIPPING",
+				"firstName"	=>	$item["firstname"],
+				"lastName"	=>	$item["lastname"],
+				"street1"	=>	$item["street"],
+				"city"		=>	$item["city"],
+				"state"		=>	$item["region"],
+				"zipCode"	=>	$item["postcode"],
+				"country"	=>	$item["country_id"],
+			);
+		}
+
+		$updateAccountPayload = array(
+			'userId'       	         => $customer->getEmail(),
+			'firstName'       	     => $customer->getFirstname(),
+			'lastName'       	     => $customer->getLastname(),
+			'email'       	         => $customer->getEmail(),
+			'profileType'       	 => "Consumer",
+			'isEmailValidated'       => false,
+			'isPhoneNumberValidated' => false,
+			'addressList'            => $AddressList,
+			'deviceContext'          => $deviceContext,
+			'_metadata'     		 => $_metadata
+		);
+		return $updateAccountPayload;
+	}
+
+	private function generateChargebackPayload($order)
+	{
+		$_metadata = array(
+			"trackingId"		=> $this->dfp->GUID(),
+			"merchantTimeStamp"	=> date('c')
+		);
+
+		$chargeBackPayload = array(
+			'chargebackId'       	=> $this->dfp->GUID(),
+			'status'       			=> 'Inquiry',
```

Comment:
I see a lot of statuses as plain strings; should these be enums so they're easy to update in the future?

Reply:
Good idea.  Here's the IDE picking it up:

![image](https://user-images.githubusercontent.com/165767/69602393-41ce3700-0fcc-11ea-8ec0-3f8e60c238b1.png)